specification/maps/resource-manager/Microsoft.Maps/stable/2023-06-01/maps-management.json

```diff
@@ -0,0 +1,1413 @@
+          "type": "string"
+        }
+      }
+    },
+    "OperationProperties": {
+      "description": "Properties of operation, include metric specifications.",
+      "type": "object",
+      "properties": {
+        "serviceSpecification": {
+          "$ref": "#/definitions/ServiceSpecification",
+          "description": "One property of operation, include metric specifications."
+        }
+      }
+    },
+    "ServiceSpecification": {
+      "description": "One property of operation, include metric specifications.",
+      "type": "object",
+      "properties": {
+        "metricSpecifications": {
+          "description": "Metric specifications of operation.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MetricSpecification"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        }
+      }
+    },
+    "MetricSpecification": {
+      "description": "Metric specification of operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of metric specification."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Display name of metric specification."
+        },
+        "displayDescription": {
+          "type": "string",
+          "description": "Display description of metric specification."
+        },
+        "unit": {
+          "type": "string",
+          "description": "Unit could be Count."
+        },
+        "dimensions": {
+          "description": "Dimensions of map account.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Dimension"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "aggregationType": {
+          "type": "string",
+          "description": "Aggregation type could be Average."
+        },
+        "fillGapWithZero": {
+          "type": "boolean",
+          "description": "The property to decide fill gap with zero or not."
+        },
+        "category": {
+          "type": "string",
+          "description": "The category this metric specification belong to, could be Capacity."
+        },
+        "resourceIdDimensionNameOverride": {
+          "type": "string",
+          "description": "Account Resource Id."
+        },
+        "sourceMdmAccount": {
+          "type": "string",
+          "description": "Source metrics account."
+        },
+        "internalMetricName": {
+          "type": "string",
+          "description": "Internal metric name."
+        },
+        "lockAggregationType": {
+          "type": "string",
+          "description": "Lock aggregation type for metrics."
+        },
+        "sourceMdmNamespace": {
+          "type": "string",
+          "description": "Metrics namespace."
+        },
+        "supportedAggregationTypes": {
+          "type": "string",
```

Comment:
>", [](http://example.com/codeflow?start=25&length=2)

consider if this can be better modelled as anenum

Reply:
I don't think we have a defined set of types because these are from Geneva. We've added these as backfill for swagger validation fixes. But I think that Geneva supports moving onboarding outside of the operations API.